src/Swift.Bindings/src/Demangler/PunyCode.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BindingsGeneration.Demangling {
+	/// <Summary>
+	/// A class for decoding puny code. Encoder has been removed.
+	/// </Summary>
+	public class PunyCode {
```

Comment:
Why is this needed demangling? Are the Swift entry point represented using punycode encoding?

Reply:
Yes. Since these are entry points in a standard .dylib, they have to be ascii encoded so they adopt puny code for anything weird.

Reply:
Do you think we could re-use the .NET `IdnMapping` implemenation for this https://learn.microsoft.com/en-us/dotnet/api/system.globalization.idnmapping?view=net-9.0 ? It has both `GetAscii` and `GetUnicode` encode and decode functions.

Reply:
The problem with this is that it's not 100% puny code. From the Apple repository:
```
// These functions implement a variant of the Punycode algorithm from RFC3492,
// originally designed for encoding international domain names, for the purpose
// of encoding Swift identifiers into mangled symbol names. This version differs
// from RFC3492 in the following respects:
// - '_' is used as the encoding delimiter instead of '-'.
// - Encoding digits are represented using [a-zA-J] instead of [a-z0-9], because
//   symbol names are case-sensitive, and Swift mangled identifiers cannot begin
//   with a digit.
// - Optionally, non-symbol ASCII characters (characters except [$_a-zA-Z0-9])
//   are mapped to the code range 0xD800 - 0xD880 and are also encoded like
//   non-ASCII unicode characters.
```

Reply:
Thank you for the detailed info. I would add a comment to the <Summary> that this is Swift specific punycode.